In [31]:
# Leitura do arquivo csv
import pandas as pd
import numpy as np

df1 = pd.read_csv("../data/TabelaCatalogo49.csv")
df2 = pd.read_csv("../data/TabelaGeralBd.csv")
print("Quantidades de linhas no df1:",len(df1))
print("Quantidades de linhas no df2:",len(df2))

Quantidades de linhas no df1: 4891
Quantidades de linhas no df2: 35752


In [39]:
print("Suportes na df1:",len(df1.suporte.unique()),df1.suporte.unique())
print("### ### ###")
print("Suportes na df2:",len(df2.suporte.unique()),df2.suporte.unique())

Suportes na df1: 24 ['Diafilmes' 'Fitas de Rolo' 'Fitas Cassete' 'Vídeo Comercial'
 'Filme 16mm' 'Vídeo Institucional' 'Microfilmes' 'Papel' 'Fotografias'
 'Diapositivo (Slides)' 'Projeto Memória' 'Microfichas'
 'Papel Fotográfico' 'Negativo em Vidro' 'Disquete' 'CD-Rom'
 'Negativo Fotográfico' 'video institucional' 'Vídeo Instiutcional' 'DVD'
 'Vídeo Terceiros' 'Hemeroteca' 'Fotografia em Vidro' nan]
### ### ###
Suportes na df2: 25 ['Filme 16mm' 'Diafilmes' 'Fitas de Rolo' 'Fitas Cassete'
 'Vídeo Comercial' 'Vídeo Institucional' 'Microfilmes' 'Tese'
 'Microfichas' 'Fotografias' 'Diapositivo (Slides)' 'Negativo Fotográfico'
 'Hemeroteca' 'Papel' 'Projeto Memória' 'Papel Fotográfico'
 'Fotografia em Vidro' 'Negativo em Vidro' 'CD-Rom' 'Disquete' nan '.'
 'tese' 'diapositivo' 'slides']


{'.',
 'DVD',
 'Tese',
 'Vídeo Instiutcional',
 'Vídeo Terceiros',
 'diapositivo',
 'slides',
 'tese',
 'video institucional'}

In [42]:
print("Suportes que não são comuns: ",set(df1.suporte.unique()) ^ set(df2.suporte.unique()))

Suportes que não são comuns:  {'.', 'Vídeo Terceiros', 'diapositivo', 'video institucional', 'Vídeo Instiutcional', 'tese', 'DVD', 'slides', 'Tese'}


In [57]:
quantidades_por_suportedf1 = df1.groupby('suporte')['suporte'].agg([np.size])
quantidades_por_suportedf1.sort_values(['size'],ascending=False,inplace=True)
print("Quantidade de itens por suporte df1 ",quantidades_por_suportedf1)

Quantidade de itens por suporte df1                        size
suporte                   
Microfilmes           1494
Fitas Cassete          752
Diapositivo (Slides)   594
Vídeo Comercial        482
Microfichas            319
Hemeroteca             281
Vídeo Institucional    226
Fotografias            211
Papel                  138
Fitas de Rolo          113
Negativo Fotográfico    75
CD-Rom                  56
Filme 16mm              43
Projeto Memória         41
Diafilmes               19
Papel Fotográfico       16
Vídeo Terceiros         13
Vídeo Instiutcional      6
DVD                      5
Fotografia em Vidro      1
Negativo em Vidro        1
Disquete                 1
video institucional      1


In [60]:
quantidades_por_suportedf2 = df2.groupby('suporte')['suporte'].agg([np.size])
quantidades_por_suportedf2.sort_values(['size'],ascending=False,inplace=True)
print("Quantidade de itens por suporte df2 ",quantidades_por_suportedf2)

Quantidade de itens por suporte df2                         size
suporte                    
Diapositivo (Slides)  24325
Tese                   7500
Microfilmes            1494
Fitas Cassete           689
Vídeo Comercial         331
Microfichas             319
Hemeroteca              282
Fotografias             209
Vídeo Institucional     161
Papel                   128
Fitas de Rolo           113
Negativo Fotográfico     68
Filme 16mm               43
Projeto Memória          39
Diafilmes                19
Papel Fotográfico        16
CD-Rom                    6
slides                    1
diapositivo               1
.                         1
Negativo em Vidro         1
Fotografia em Vidro       1
Disquete                  1
tese                      1


## Observações

- Criar o tipo de conteúdo no site caph.fflch.usp.br com os campos pertinentes
- não importar ainda, pois vamos reversar o nids para cada típo
- Atenção: Lembrar de checar duplicados pelo TOMBO nos dois arquivos csv
- Jpa foi feito: Tese                   7500 - Já foi feito

## Ricardo Fernando dos Santos

- Diapositivo (Slides)  24325 + Diapositivo (Slides)   594
- Fotografias             209 + Fotografias            211
- Papel Fotográfico        16 + Papel Fotográfico       16

## Atenágoras Souza Silva

- Diafilmes                19 + Diafilmes               19
- Microfilmes            1494 + Microfilmes           1494
- Fitas Cassete           689 + Fitas Cassete          752

##  Marcos Vinícius Andrade de Moura

- Microfichas             319 + Microfichas            319
- Hemeroteca              282 + Hemeroteca             281

## Juliana Saotome

- Vídeo Comercial         331 + Vídeo Comercial        482
- Vídeo Institucional     161 + Vídeo Institucional    226
- Vídeo Terceiros         13
- Vídeo Instiutcional      6

## Letícia Paulo Almeida

- Papel                   128 + Papel                  138
- Fitas de Rolo           113 + Fitas de Rolo          113
- Negativo Fotográfico     68 + Negativo Fotográfico    75
- Filme 16mm               43 + Filme 16mm              43

## Rodrigo Aroni Siquette

- CD-Rom                  56 + CD-Rom                    6
- Projeto Memória         41 + Projeto Memória          39

## Roger Luiz Bolsi

Copiar conjuntos documentais
http://d7caph.fflch.usp.br/conjunto-documental




